File: CustomizableButtonsModule/src/main/java/com/github/gabrielgouv/customizablebuttons/DimensionUtil.java
package com.github.gabrielgouv.customizablebuttons;

import android.content.res.Resources;


public final class DimensionUtil {

    public static int pxToDip(float px) {
        return (int) (px / Resources.getSystem().getDisplayMetrics().density);
    }

    public static int dipToPx(float dip) {
        return (int) (dip * Resources.getSystem().getDisplayMetrics().density);
    }

    public static int pxToSp(float px) {
        return (int) (px / Resources.getSystem().getDisplayMetrics().scaledDensity);
    }

    public static int spToPx(final float sp) {
        return (int) (sp * Resources.getSystem().getDisplayMetrics().scaledDensity);
    }

}

File: CustomizableButtonsModule/src/main/java/com/github/gabrielgouv/customizablebuttons/util/DimensionUtil.kt
package com.github.gabrielgouv.customizablebuttons.util

import android.content.res.Resources

object DimensionUtil {

    fun pxToDip(px: Float): Int {
        return (px / Resources.getSystem().displayMetrics.density).toInt()
    }

    fun dipToPx(dip: Float): Int {
        return (dip * Resources.getSystem().displayMetrics.density).toInt()
    }

    fun pxToSp(px: Float): Int {
        return (px / Resources.getSystem().displayMetrics.scaledDensity).toInt()
    }

    fun spToPx(sp: Float): Int {
        return (sp * Resources.getSystem().displayMetrics.scaledDensity).toInt()
    }

}

